luigi/parameter.py

```diff
@@ -995,3 +996,74 @@ def parse(self, s):
                 raise OSError("could not make parent directories for {s}".format(
                     s=s))
         return s
+
+
+class NumericalParameter(Parameter):
+    """
+    Parameter whose value is a number of the specified type, e.g. ``int`` or
+    ``float`` and in the range specified.
+
+    In the task definition, use
+
+    .. code-block:: python
+
+        class MyTask(luigi.Task):
+            my_param = luigi.NumericalParameter(
+                var_type=int, min_value=-3, max_value=7, left_op=operator.le, right_op=operator.lt)
```

Comment:
Yeah, I think you're right - this sort of just grew organically from a particular use case I had where the way I implemented were sensible defaults (autoincrement database IDs).  I have no attachment to the defaults as they are so will go with yours, as they're more natural assumptions given the language.